alf/algorithms/algorithm.py

```diff
@@ -245,24 +242,25 @@ def _set_exp_replayer(self, sample_exp):
         Args:
             sample_exp (nested Tensor):
         """
+        if (self._exp_replayer_num_envs is None
+                or self._exp_replayer_length is None
+                or self._prioritized_sampling is None):
+            raise RuntimeError(
+                'Experience replayer cannot be initialized before set_exp_replayer() is called'
```

Comment:
"Experience replayer must be initialized first by calling ``set_exp_replayer()`` before ``observe_for_replay()`` is called!"

Reply:
Updated the message.